app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -826,12 +827,19 @@ public void onFeedListChanged(FeedListUpdateEvent event) {
     private void handleNavIntent() {
         Log.d(TAG, "handleNavIntent()");
         Intent intent = getIntent();
-        if (intent.hasExtra(EXTRA_FEED_ID) ||
+        if (intent.hasExtra(EXTRA_FEED_ID) || intent.hasExtra(EXTRA_REFRESH_ON_START) ||
                 (intent.hasExtra(EXTRA_NAV_TYPE) &&
                         (intent.hasExtra(EXTRA_NAV_INDEX) || intent.hasExtra(EXTRA_FRAGMENT_TAG)))) {
             int index = intent.getIntExtra(EXTRA_NAV_INDEX, -1);
             String tag = intent.getStringExtra(EXTRA_FRAGMENT_TAG);
             Bundle args = intent.getBundleExtra(EXTRA_FRAGMENT_ARGS);
+            boolean refreshOnStart = intent.getBooleanExtra(EXTRA_REFRESH_ON_START, false);
+            if (refreshOnStart) {
+                tag = QueueFragment.TAG;
+                args = new Bundle();
+                args.putBoolean(QueueFragment.ARG_REFRESH_ON_START, true);
```

Comment:
Would it be possible to just call `AutoUpdateManager.runImmediate(requireContext());` here directly instead of passing it to the queue?

Reply:
@ByteHamster 

That's would be better, and fixed at 1167958.

Some fragments have menu item for refresh, and that is started to animate when app starts via shortcut.

```xml
    <item
        android:id="@+id/refresh_item"
        android:title="@string/refresh_label"
        android:menuCategory="container"
        custom:showAsAction="always"
        android:icon="?attr/navigation_refresh"/>
```

But some screens have no menu item for that, like `PlaybackHistoryFragment`.
When app has been started already and the last screen have no menu item,  user cannot know that app is refreshing all feeds in background.

Showing toast or something might be helpful to know that.